internal/common/constant/constant.go

```diff
@@ -52,3 +57,16 @@ const TrueString = "True"
 // MaximumNacObjectNameLength represents Generated Non Admin Object Name and
 // must be below 63 characters, because it's used within object Label Value
 const MaximumNacObjectNameLength = validation.DNS1123LabelMaxLength
+
+// NonAdminCondition are used for more detailed information supporting NonAdminPhase state.
+type NonAdminCondition string
```

Comment:
files in `api/v1alpha1` were created by kubebuilder

The file it was stored was created by us

I want to avoid compability issues with kubebuilder

Reply:
> I really don't know how our additions to the generated by the operator-sdk code could be problematic. That is the reason to use it, to generate scaffold and then we build on top of it? 

Yes, we can code on top of things, but if we create a CRD called "NonAdminController" scaffold will break becasue https://github.com/migtools/oadp-non-admin/blob/master/api/v1alpha1/nonadmincontroller_types.go already exists

> Conditions and Phases are integral part of an API, aren't they?

more or less

we use `NonAdminCondition` type for populating NonAdminBackup/Restore `status.conditions.[].type`, but that field is not enforced to use that type https://github.com/migtools/oadp-non-admin/blob/master/config/crd/bases/oadp.openshift.io_nonadminbackups.yaml#L595-L604 (but phase is indeed typed :heavy_check_mark:)

If you go on master branch and delete this line https://github.com/migtools/oadp-non-admin/blob/master/api/v1alpha1/nonadmincontroller_types.go#L21 you will see nothing breaks. We only need the consts